pkg/kubectl/cmd/run.go

```diff
@@ -106,6 +107,23 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 		return cmdutil.UsageError(cmd, "NAME is required for run")
 	}
 
+	// Let kubectl run follow rules for `--`
+	var commandAndCustomArgs []string
```

Comment:
Nm, I reminded myself.  There may be a more succint way to fix this

1. `kubectl run -p POD` should have zero args
2. `kubectl exec POD` should have zero args
3. `kubectl run POD COMMAND` should have 1 arg
4. `kubectl run -p POD COMMAND` should have 1 arg

If you use SetInterspersed on the cobra.Command object, does that give you the correct behavior in all of these cases?  (you might have to grab arg[0] for case 3 and shorten args to args[1:])